improver/utilities/call_object_method.py

```diff
@@ -0,0 +1,31 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+
+"""module to give access to callable methods on objects."""
+
+
+def call_object_method(obj: object, method_name: str, **kwargs):
```

Comment:
```suggestion
def call_cubelike_method(*cubes: Union[Cube, CubeList], method_name: str = None, **kwargs): -> Union[Cube, CubeList]
```

Reply:
My thinking was that we did not have to be restricted to cubelike objects. ANY object could be used in this way, e.g. a NumPy array or a Pandas dataframe.

I am wondering whether this would be better as a Plugin class though, with the init method taking the name of the attribute to be called, so that args and kwargs can be disambiguated.

Reply:
I've also realised that we don't need this at all. For any class object, you can do `MyClass.method(class_object)`, so `iris.cube.Cube.collapsed(temperature_cube, "height", iris.analysis.MAX)` would give the same answer as `temperature_cube.collapsed("height", iris.analysis.MAX)`, so I don't think this PR adds anything we can't already do.